lib/routes/deepin/thread.ts

```diff
@@ -0,0 +1,99 @@
+import { Route } from '@/types';
+import ofetch from '@/utils/ofetch';
+import cache from '@/utils/cache';
+import { parseDate } from '@/utils/parse-date';
+
+export const route: Route = {
+    path: '/threads/:type?',
+    categories: ['bbs'],
+    example: '/deepin/threads/latest',
+    parameters: {
+        type: {
+            description: '主题类型',
+            options: [
+                {
+                    value: 'hot',
+                    label: '最热主题',
+                },
+                {
+                    value: 'latest',
+                    label: '最新主题',
+                },
+            ],
+        },
+    },
+    name: '首页主题列表',
+    maintainers: ['myml'],
+    radar: [
+        {
+            source: ['bbs.deepin.org'],
+            target: '/threads/latest',
+        },
+    ],
+    handler,
+};
+
+interface ThreadIndexResult {
+    ThreadIndex: {
+        id: number;
+        subject: string;
+        created_at: string;
+        user: { nickname: string };
+        forum: { name: string };
+    }[];
+}
+interface ThreadInfoResult {
+    data: {
+        id: number;
+        subject: string;
+        created_at: string;
+        user: { nickname: string };
+        post: { message: string };
+    };
+}
+
+const TypeMap = {
+    hot: { where: 'hot_value', title: '最热主题' },
+    latest: { where: 'id', title: '最新主题' },
+};
+
+async function handler(ctx) {
+    let type = TypeMap.latest;
+    if (ctx.req.param('type') === 'hot') {
+        type = TypeMap.hot;
+    }
+    const res = await ofetch<ThreadIndexResult>('https://bbs.deepin.org.cn/api/v1/thread/index', {
+        query: {
+            languages: 'zh_CN',
+            order: 'updated_at',
+            where: type.where,
+        },
+        headers: {
+            accept: 'application/json',
+        },
+    });
+    const items = await Promise.all(
+        res.ThreadIndex.map(async (thread) => {
+            const item = {
+                id: String(thread.id),
+                title: thread.subject,
+                link: 'https://bbs.deepin.org.cn/post/' + thread.id,
+                pubDate: parseDate(thread.created_at),
+                author: thread.user.nickname,
+                category: [thread.forum.name],
+                description: '',
+            };
+            const cacheData = await cache.tryGet(item.link, () => ofetch<ThreadInfoResult>('https://bbs.deepin.org.cn/api/v1/thread/info?id=' + item.id));
```

Comment:
https://github.com/DIYgod/RSSHub/pull/16818#discussion_r1767022842

Please cache the whole returned object of https://github.com/DIYgod/RSSHub/blob/f16687f2160503462a8d3994cfa2fea609250612/lib/routes/deepin/thread.ts#L91

instead of their API response

Reply:
The type returned by cache.tryGet is Promise<string | Record<string, any> | null> ,
Route's handler type is (ctx: Context) => Promise<Data> | Data

If you cache the entire returned object, typescript will complain about type mismatch.

Reply:
Do you mean to convert the return value of tryGet , Promise<string | Record<string, any> | null> , to the type of Route.handler , 'Promise<Data> | Data' ?

I don't want to use "as" and I don't understand why it's better to force type conversion by caching the whole object instead.

Reply:
I mean caching this whole returned object of https://github.com/DIYgod/RSSHub/blob/f16687f2160503462a8d3994cfa2fea609250612/lib/routes/deepin/thread.ts#L91 which consist of
https://github.com/DIYgod/RSSHub/blob/f16687f2160503462a8d3994cfa2fea609250612/lib/routes/deepin/thread.ts#L77-L85

and casting the returned cache it as [`Promise<DataItem>`](https://github.com/DIYgod/RSSHub/blob/003e3adef9abc7374787720b8d8868329a075ddd/lib/types.ts#L32)

Reply:
Updated PR with requested changes.